tests/test_bias.py

```diff
@@ -0,0 +1,168 @@
+
+    def test_independentFrameLevel(self):
+        """Test image mean.
+
+        Notes
+        -----
+        DMTN-101 4.2
+
+        Confirm that the mean of the result is 0 to within statistical
+        error
+        """
+        mean = afwMath.makeStatistics(self.exposure.getImage(), afwMath.MEAN).getValue()
+        self.assertLess(np.abs(mean), 1, msg=f"Test 4.2: {mean}")
+
+    def test_independentFrameSigma(self):
+        """Amp sigma against readnoise
+
+        Notes
+        -----
+        DMTN-101 4.3
+
+        Confirm that the 5-sigma clipped standard deviation of each
+        amplifier is within 5% of the nominal readnoise, as determined
+        by a robust measure of the noise in the serial overscan
+
+        """
+        ccd = self.exposure.getDetector()
+        for amp in ccd:
+            ampExposure = self.exposure.Factory(self.exposure, amp.getBBox())
+            statControl = afwMath.StatisticsControl(5.0, 5)
+            statControl.setAndMask(self.exposure.mask.getPlaneBitMask(["SAT", "BAD", "NO_DATA"]))
+            sigma = afwMath.makeStatistics(ampExposure.getImage(),
+                                           afwMath.STDEVCLIP, statControl).getValue()
+            # needs to be < 0.05
+            fractionalError = np.abs(sigma - amp.getReadNoise())/amp.getReadNoise()
+            self.assertLess(fractionalError, 0.71, msg=f"Test 4.3: {amp.getName()} {fractionalError}")
+
+    def test_amplifierSigma(self):
+        """Clipped sigma against CR-rejected sigma
+
+        Notes
+        -----
+        DMTN-101 4.4
+
+        Run a CR rejection on the result and confirm that the
+        unclipped standard deviation is consistent with the 5-sigma
+        clipped value.
+
+        """
+        crTask = RepairTask()
+        crRejected = self.exposure.clone()
+        psf = measAlg.SingleGaussianPsf(21, 21, 3.0)
+        crRejected.setPsf(psf)
+        crTask.run(crRejected, keepCRs=False)
+
+        ccd = self.exposure.getDetector()
+        for amp in ccd:
+            ampExposure = self.exposure.Factory(self.exposure, amp.getBBox())
+            clipControl = afwMath.StatisticsControl(5.0, 5)
+            clipControl.setAndMask(self.exposure.mask.getPlaneBitMask(["SAT", "BAD", "NO_DATA"]))
+            sigmaClip = afwMath.makeStatistics(ampExposure.getImage(),
+                                               afwMath.STDEVCLIP, clipControl).getValue()
+
+            crAmp = crRejected.Factory(crRejected, amp.getBBox())
+            statControl = afwMath.StatisticsControl()
+            statControl.setAndMask(self.exposure.mask.getPlaneBitMask(["SAT", "BAD", "NO_DATA", "CR"]))
+            sigma = afwMath.makeStatistics(crAmp.getImage(), afwMath.STDEV, statControl).getValue()
+
+            # needs to be < 0.05
+            fractionalError = np.abs(sigma - sigmaClip)/sigmaClip
+            self.assertLess(fractionalError, 3.0, msg="Test 4.4: {amp.getName()} {fractionalError}")
```

Comment:
3?!

Reply:
Hmmm... That seems quite bad, but I guess this is a question of test data at some level too, perhaps, plus we've already got something above to review all the various threshold values, so let's file this under that too.

Maybe a ticket to capture the work?

Also, what do you think about pulling these out into at least a bunch of all-caps const-style VARIABLES that live at the top of the file so that we can at least see all the test values in one place, and not have them hiding as hard-coded values in each test? Or do you think that will be more messy, and just the fact that they live in the asserts in each test makes finding them easy enough?

Reply:
I've been searching them by finding the assert.  I was assuming DM-26396 will be the ticket that all the test fixes live on.  I've also thought up a possibly over-designed system of putting the test values in the testdata package (probably in yaml as well), with an override parameter for known failures.  Then the logic is that a test with `override=False` must meet the DMTN limits.  However, one with `override=True` can fail the DMTN limit as long as it meets the testdata's limit.  This separates "validation and verification" from "continuous integration" tests.

Reply:
OK, that certainly sounds interesting, and (unsurprisingly) like you're well ahead of the game here 🙂